lib/task_helpers.rb

```diff
@@ -1,213 +1,106 @@
 require 'elasticsearch'
+require_relative '../models/comment'
+require_relative '../models/comment_thread'
 
 module TaskHelpers
   module ElasticsearchHelper
     LOG = Logger.new(STDERR)
+    INDEX_MODELS = [Comment, CommentThread].freeze
+    INDEX_NAMES = [Comment.index_name, CommentThread.index_name].freeze
 
-    # Creates a new index and loads data from the database.  If an alias name
-    # is supplied, it will be pointed to the new index and catch up will be
-    # called both before and after the alias switch..
-    #
-    # Returns the name of the newly created index.
+    # Creates new indices and loads data from the database.
     #
     # Params:
-    # +alias_name+:: (optional) The alias to point to the new index.
     # +batch_size+:: (optional) The number of elements to index at a time. Defaults to 500.
-    # +sleep_time+:: (optional) The number of seconds to sleep between batches. Defaults to 0.
     # +extra_catchup_minutes+:: (optional) The number of extra minutes to catchup. Defaults to 5.
-    def self.rebuild_index(alias_name=nil, batch_size=500, sleep_time=0, extra_catchup_minutes=5)
-      initial_start_time = Time.now
-      index_name = create_index()
-
-      [Comment, CommentThread].each do |model|
+    def self.rebuild_indices(batch_size=500, extra_catchup_minutes=5)
+      create_indices
+      INDEX_MODELS.each do |model|
         current_batch = 1
-        model.import(index: index_name, batch_size: batch_size) do |response|
-            batch_import_post_process(response, current_batch, sleep_time)
+        model.import(index: model.index_name, batch_size: batch_size) do |response|
+            batch_import_post_process(response, current_batch)
             current_batch += 1
         end
       end
 
-      if alias_name
-        # Just in case initial rebuild took days and first catch up takes hours,
-        # we catch up once before the alias move and once afterwards.
-        first_catchup_start_time = Time.now
-        adjusted_start_time = initial_start_time - (extra_catchup_minutes * 60)
-        catchup_index(adjusted_start_time, index_name, batch_size, sleep_time)
+      first_catchup_start_time = Time.now
```

Comment:
For what it's worth, moving the start time down below `create_indices` and the loop over the `INDEX_MODELS` will presumably make it seem later than it really was... was there a reason to change `initial_start_time` to `first_catchup_start_time`?  It this more accurate to the spirit of what it was for, or something?

Reply:
Thanks, you are right. Done